pyam/core.py

```diff
@@ -1132,7 +1136,45 @@ def _apply_filters(data, meta, filters):
             keep_col = pattern_match(data[col], values, level, regexp)
 
         elif col == 'year':
-            keep_col = years_match(data[col], values)
+            try:
+                keep_col = years_match(data[col], values)
+            except KeyError as exc:
+                if str(exc) != "'year'":
+                    raise exc
+                elif 'time' in data.columns:
```

Comment:
Better to use `if self.time_col is not ‘time’`

Reply:
that would be great except we don't have access to `self` in `_apply_filters`, thoughts?

Reply:
right...

would it make sense to pass `self.time_col` as a required arg through `_apply_filters`, or is that making it needlessly complicated?

Reply:
I'm a big fan of trying to make these internal conventions as clear as possible. Having said that, I think my refactoring would be to actually move `_apply_filters` so it was a method of `IamDataFrame`. The reason being that `_apply_filters` is specific enough in what it expects (in particular the pattern matching) that it only makes sense to apply it to `IamDataFrame` instances, not a general dataframe. So I think options are:

1. leave it
1. pass time_col as required arg
1. refactor and move `_apply_filters` so it's a method of `IamDataFrame` (my preference but also the most complicated)

Reply:
ok I'll do it that way then (when I get around to it..)